backend/src/test/java/com/benine/backend/CameraControlTest.java

```diff
@@ -0,0 +1,188 @@
+                                            Integer.parseInt(mainConfig.getValue("serverport")));
+    // Setup camerahandler
+    cameraController = new CameraController();
+      
+    IPCamera mockcamera = mock(IPCamera.class);
+    when(mockcamera.getStreamLink()).thenReturn("http://83.128.144.84:88/cgi-bin/CGIProxy.fcgi?cmd=snapPicture&usr=user&pwd=geheim");
+    when(mockcamera.getId()).thenReturn(1);
+    when(mockcamera.getFocusPos()).thenReturn(40);
+    when(mockcamera.getIrisPos()).thenReturn(60);
+    when(mockcamera.getPosition()).thenReturn(new Position(10, 20));
+    when(mockcamera.getZoomPosition()).thenReturn(30);
+    
+    //Define which command to send when zoomTo is called on mocked camera.
+    doAnswer(new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws IpcameraConnectionException {
+          Object[] args = invocation.getArguments();
+          int zoom = (int) args[0];
+          if (zoom > 60) {
+            sendCommand("zoomIn", "83.128.144.84:88");
+          } else if( zoom < 40) {
+            sendCommand("zoomOut", "83.128.144.84:88");
+          } else {
+            sendCommand("zoomStop", "83.128.144.84:88");
+          }
+          return null;
+      }
+    }).when(mockcamera).zoomTo(any(Integer.class));
+    
+    //Define which command to send when move is called on mocked camera.
+    doAnswer(new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws IpcameraConnectionException {
+          Object[] args = invocation.getArguments();
+          int pan = (int) args[0];
+          int tilt = (int) args[1];
+          if (pan > 85) {
+            sendCommand("ptzMoveRight", "83.128.144.84:88");
+          } else if( pan < 15) {
+            sendCommand("ptzMoveLeft", "83.128.144.84:88");
+          } else if( tilt > 85) {
+            sendCommand("ptzMoveUp", "83.128.144.84:88");
+          } else if( tilt < 15) {
+            sendCommand("ptzMoveDown", "83.128.144.84:88");
+          } else {
+            sendCommand("ptzStopRun", "83.128.144.84:88");
+          }
+          return null;
+      }
+    }).when(mockcamera).move(any(Integer.class), any(Integer.class));
+    
+    when(mockcamera.toJSON()).thenReturn(createJSON());
+    cameraController.addCamera(mockcamera);
+    
+    //Add two extra camera's.
+    SimpleCamera camera2 = new SimpleCamera();
+    camera2.setStreamLink(mainConfig.getValue("camera1"));
+    cameraController.addCamera(camera2);
+    SimpleCamera camera3 = new SimpleCamera();
+    camera3.setStreamLink(mainConfig.getValue("camera1"));
+    cameraController.addCamera(camera3);
+  }
+  
+  /**
+   * This methods starts a server used to manually test the user interface.
+   * @throws InterruptedException when the thread is interrupted.
+   * @throws IOException when the server can not be created.
+   */
+  @Test
+  public void ManualTestUI() throws InterruptedException, IOException{
```

Comment:
When this is merged in develop and the httpUpdate is merged in develop, we'll have to change this to a HttpController which makes this a class (and especially this part) a lot cleaner.